internal/controller/deployment.go

```diff
@@ -39,11 +40,35 @@ func constructRuntimeConfigSecretMount(_ctx context.Context, secretName string)
 	return volume, volumeMount
 }
 
+func constructCASecretMount(_ context.Context, caSecretName string) (corev1.Volume, corev1.VolumeMount) {
+	volume := corev1.Volume{
+		Name: "spin-ca",
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: caSecretName,
+				Optional:   ptr(true),
+				Items: []corev1.KeyToPath{{
+					Key:  "ca-certificates.crt",
+					Path: "ca-certificates.crt",
+				}},
+			},
+		},
+	}
+	volumeMount := corev1.VolumeMount{
+		Name:      "spin-ca",
+		ReadOnly:  true,
+		MountPath: "/etc/ssl/certs/ca-certificates.crt",
+		SubPath:   "ca-certificates.crt",
+	}
+
+	return volume, volumeMount
+}
+
 // ConstructVolumeMountsForApp introspects the application and generates
 // any required volume mounts. A generated runtime secret is mutually
 // exclusive with a user-provided secret - this is to require _either_ a
 // manual runtime-config or a generated one from the crd.
-func ConstructVolumeMountsForApp(ctx context.Context, app *spinv1alpha1.SpinApp, generatedRuntimeSecret string) ([]corev1.Volume, []corev1.VolumeMount, error) {
+func ConstructVolumeMountsForApp(ctx context.Context, app *spinv1alpha1.SpinApp, generatedRuntimeSecret, caSecretName string) ([]corev1.Volume, []corev1.VolumeMount, error) {
```

Comment:
I wish there was a way to make `caSecretName` use an `Option<string>` instead of an empty string to make it more clear to the caller that this is an optional value. I've been spoiled by Rust.

Reply:
we can always use *string in golang. (before you ask Even I am not sure if I am kidding or being serious here.)